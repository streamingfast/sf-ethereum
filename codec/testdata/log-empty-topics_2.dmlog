FIRE INIT 2.3 geth 1.13.15-fh2.4-db154a58-2024041
FIRE BEGIN_BLOCK 3325385
FIRE BEGIN_APPLY_TRX 7f5a91a9036683dc4e3f75f914f2c78bac0509e7add3a428db560863102ae028 702641c70a11e480f646ed247d078c65abaac5de . 0bfa 51cf04f9e7adcc2c6284005fb3cc4a9780f630996e84bd5c3f9bd62b26710741 02f5a60ac274895bf0393756d1a7afe28d2ba61426ac8943983454604ba6b18f 95000 0ba43b7400 23 2bed2182 00 . . 0 1 3
FIRE TRX_FROM 821b55d8abe79bc98f05eb675fdc50dfe796b7ab
FIRE BALANCE_CHANGE 0 821b55d8abe79bc98f05eb675fdc50dfe796b7ab 02863c1f5cdae42f953fff9e50 02863c1f5cdad34f7bb150be50 gas_buy 2
FIRE GAS_CHANGE 0 95000 73936 intrinsic_gas 3
FIRE NONCE_CHANGE 0 821b55d8abe79bc98f05eb675fdc50dfe796b7ab 23 24 4
FIRE EVM_RUN_CALL CALL 1 5
FIRE EVM_PARAM CALL 1 821b55d8abe79bc98f05eb675fdc50dfe796b7ab 702641c70a11e480f646ed247d078c65abaac5de . 73936 2bed2182
FIRE GAS_CHANGE 1 73772 73397 event_log 6
FIRE ADD_LOG 1 0 702641c70a11e480f646ed247d078c65abaac5de  . 7
FIRE EVM_END_CALL 1 73388 . 8
FIRE BALANCE_CHANGE 0 821b55d8abe79bc98f05eb675fdc50dfe796b7ab 02863c1f5cdad34f7bb150be50 02863c1f5cdae058c874daae50 gas_refund 9
FIRE BALANCE_CHANGE 0 821b55d8abe79bc98f05eb675fdc50dfe796b7ab 02863c1f5cdae058c874daae50 02863c1f5cdae42f953fff9e50 reward_transaction_fee 10
FIRE END_APPLY_TRX 21612 . 100934 00000000000000000000000000000000000000000000000000000000000000000000000000000000000000008000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000004000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000002000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000 11 [{"address":"0x702641c70a11e480f646ed247d078c65abaac5de","data":"0x","topics":[]}]
FIRE FINALIZE_BLOCK 3325385
FIRE END_BLOCK 3325385 3575 {"finalizedBlockHash":"0x7406739b969f95b1a38c94dbce60594ebda5b751ef8bb116181bbdd443513109","finalizedBlockNum":"0x32bdc9","header":{"parentHash":"0x8eededb3ef47265acb8bd4faeca40dbc0b19773667a9cc86ea45f7113e2c459f","sha3Uncles":"0x1dcc4de8dec75d7aab85b567b6ccd41ad312451b948a7413f0a142fd40d49347","miner":"0x9a6034c84cd431409ac1a35278c7da36ffda53e5","stateRoot":"0x7ee881ba39d4664ab5743b8575664bba86b76683aae9b4dd5ccc605dfe0c67f5","transactionsRoot":"0xd3f5a653a01d19315b1534125eb627a19aaf8fc60c8dc86f2e111f67aca447ca","receiptsRoot":"0x3c3d95195f9ea78b1ee5c7104db0713c8b5a0c72e98f0fe03dd49f48a551f7fe","logsBloom":"0x04200000004000008000000400000000000080000000000040000040000000080000000000000000000080000000060000001000000000000400000800040800001000000000000000800008000000000000000000048004000040000000000000000000020000000000800002000800000000020000400000000010000000082200100000000000000000002804000001000400000008008044000000000000008000000000000000000500000000002002000000002000000000004400400000000002000180000000000000000000010000000000000000000080020060000200000000000000000000000000000000210000008000000000000000000000","difficulty":"0x0","number":"0x32bdc9","gasLimit":"0x1c9c380","gasUsed":"0xb0614","timestamp":"0x6441081c","extraData":"0xd883010b05846765746888676f312e32302e32856c696e7578","mixHash":"0x3b19b0099cb461b256793c359ec1037079703e45460788d02b4e1cbae079546b","nonce":"0x0000000000000000","baseFeePerGas":"0x7","withdrawalsRoot":"0x769f72989ccf56ae46c1be2117dc41847a02bfc1e5eefd5f740f92def5fa257b","blobGasUsed":null,"excessBlobGas":null,"parentBeaconBlockRoot":null,"hash":"0x7406739b969f95b1a38c94dbce60594ebda5b751ef8bb116181bbdd443513109"},"totalDifficulty":"0x3c656d23029ab0","uncles":[]}
