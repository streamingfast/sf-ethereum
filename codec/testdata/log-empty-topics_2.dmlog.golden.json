[
  {
    "hash": "7406739b969f95b1a38c94dbce60594ebda5b751ef8bb116181bbdd443513109",
    "header": {
      "baseFeePerGas": {
        "bytes": "07"
      },
      "coinbase": "9a6034c84cd431409ac1a35278c7da36ffda53e5",
      "difficulty": {
        "bytes": "00"
      },
      "extraData": "d883010b05846765746888676f312e32302e32856c696e7578",
      "gasLimit": "30000000",
      "gasUsed": "722452",
      "hash": "7406739b969f95b1a38c94dbce60594ebda5b751ef8bb116181bbdd443513109",
      "logsBloom": "04200000004000008000000400000000000080000000000040000040000000080000000000000000000080000000060000001000000000000400000800040800001000000000000000800008000000000000000000048004000040000000000000000000020000000000800002000800000000020000400000000010000000082200100000000000000000002804000001000400000008008044000000000000008000000000000000000500000000002002000000002000000000004400400000000002000180000000000000000000010000000000000000000080020060000200000000000000000000000000000000210000008000000000000000000000",
      "mixHash": "3b19b0099cb461b256793c359ec1037079703e45460788d02b4e1cbae079546b",
      "number": "3325385",
      "parentHash": "8eededb3ef47265acb8bd4faeca40dbc0b19773667a9cc86ea45f7113e2c459f",
      "receiptRoot": "3c3d95195f9ea78b1ee5c7104db0713c8b5a0c72e98f0fe03dd49f48a551f7fe",
      "stateRoot": "7ee881ba39d4664ab5743b8575664bba86b76683aae9b4dd5ccc605dfe0c67f5",
      "timestamp": "2023-04-20T09:38:36Z",
      "totalDifficulty": {
        "bytes": "3c656d23029ab0"
      },
      "transactionsRoot": "d3f5a653a01d19315b1534125eb627a19aaf8fc60c8dc86f2e111f67aca447ca",
      "uncleHash": "1dcc4de8dec75d7aab85b567b6ccd41ad312451b948a7413f0a142fd40d49347",
      "withdrawalsRoot": "769f72989ccf56ae46c1be2117dc41847a02bfc1e5eefd5f740f92def5fa257b"
    },
    "number": "3325385",
    "size": "3575",
    "transactionTraces": [
      {
        "beginOrdinal": "1",
        "calls": [
          {
            "address": "702641c70a11e480f646ed247d078c65abaac5de",
            "balanceChanges": [
              {
                "address": "821b55d8abe79bc98f05eb675fdc50dfe796b7ab",
                "newValue": {
                  "bytes": "02863c1f5cdad34f7bb150be50"
                },
                "oldValue": {
                  "bytes": "02863c1f5cdae42f953fff9e50"
                },
                "ordinal": "2",
                "reason": "REASON_GAS_BUY"
              },
              {
                "address": "821b55d8abe79bc98f05eb675fdc50dfe796b7ab",
                "newValue": {
                  "bytes": "02863c1f5cdae058c874daae50"
                },
                "oldValue": {
                  "bytes": "02863c1f5cdad34f7bb150be50"
                },
                "ordinal": "9",
                "reason": "REASON_GAS_REFUND"
              },
              {
                "address": "821b55d8abe79bc98f05eb675fdc50dfe796b7ab",
                "newValue": {
                  "bytes": "02863c1f5cdae42f953fff9e50"
                },
                "oldValue": {
                  "bytes": "02863c1f5cdae058c874daae50"
                },
                "ordinal": "10",
                "reason": "REASON_REWARD_TRANSACTION_FEE"
              }
            ],
            "callType": "CALL",
            "caller": "821b55d8abe79bc98f05eb675fdc50dfe796b7ab",
            "endOrdinal": "8",
            "executedCode": true,
            "gasChanges": [
              {
                "newValue": "73936",
                "oldValue": "95000",
                "ordinal": "3",
                "reason": "REASON_INTRINSIC_GAS"
              },
              {
                "newValue": "73397",
                "oldValue": "73772",
                "ordinal": "6",
                "reason": "REASON_EVENT_LOG"
              }
            ],
            "gasConsumed": "548",
            "gasLimit": "73936",
            "index": 1,
            "input": "2bed2182",
            "logs": [
              {
                "address": "702641c70a11e480f646ed247d078c65abaac5de",
                "ordinal": "7"
              }
            ],
            "nonceChanges": [
              {
                "address": "821b55d8abe79bc98f05eb675fdc50dfe796b7ab",
                "newValue": "24",
                "oldValue": "23",
                "ordinal": "4"
              }
            ]
          }
        ],
        "endOrdinal": "11",
        "from": "821b55d8abe79bc98f05eb675fdc50dfe796b7ab",
        "gasLimit": "95000",
        "gasPrice": {
          "bytes": "0ba43b7400"
        },
        "gasUsed": "21612",
        "hash": "7f5a91a9036683dc4e3f75f914f2c78bac0509e7add3a428db560863102ae028",
        "index": 3,
        "input": "2bed2182",
        "nonce": "23",
        "r": "51cf04f9e7adcc2c6284005fb3cc4a9780f630996e84bd5c3f9bd62b26710741",
        "receipt": {
          "cumulativeGasUsed": "100934",
          "logs": [
            {
              "address": "702641c70a11e480f646ed247d078c65abaac5de",
              "ordinal": "7"
            }
          ],
          "logsBloom": "00000000000000000000000000000000000000000000000000000000000000000000000000000000000000008000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000004000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000002000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000"
        },
        "s": "02f5a60ac274895bf0393756d1a7afe28d2ba61426ac8943983454604ba6b18f",
        "status": "SUCCEEDED",
        "to": "702641c70a11e480f646ed247d078c65abaac5de",
        "v": "0bfa"
      }
    ],
    "ver": 3
  }
]